File: src/main/java/com/zerozero/store/domain/value/Address.java
package com.zerozero.store.domain.value;

import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Embeddable
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Address {
    private String address;
    private String roadAddress;

    public static Address of(String address, String roadAddress) {
        return new Address(address, roadAddress);
    }
}

File: src/main/java/com/zerozero/store/domain/value/Address.kt
package com.zerozero.store.domain.value

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
data class Address(
    @Column(name = "address", nullable = false)
    val address: String,

    @Column(name = "road_address", nullable = false)
    val roadAddress: String
)
